CouchBaseStorage/Configuration/OrleansConfigurationExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Configuration;
+using CouchBaseProviders.Configuration.OrleansDocumentExpiry;
+
+namespace CouchBaseProviders.Configuration
+{
+    /// <summary>
+    /// Orleans configuration utility extension methods.
+    /// </summary>
+    public static class OrleansConfigurationExtensions
+    {
+        public static Dictionary<string, TimeSpan> ReadDocumentExpiryConfiguration()
+        {
+            var result = new Dictionary<string, TimeSpan>();
+
+            if (string.IsNullOrWhiteSpace(OrleansGrainExpiryConstants.OrleansConfigurationSectionPath)) return result;
+
+            var grainExpiriesConfig = (OrleansConfigurationSection)ConfigurationManager.GetSection(OrleansGrainExpiryConstants.OrleansConfigurationSectionPath);
+            if (grainExpiriesConfig == null || grainExpiriesConfig.GrainExpiries.Count == 0) return result;
+
+            foreach (GrainExpiry documentExpiry in grainExpiriesConfig.GrainExpiries)
+            {
+                var expiry = TimeSpan.Parse(documentExpiry.Expiry);
```

Comment:
I think we should use `TimeSpan.TryParse` or wrap in try/catch blocks to prevent an exception bubbling out if parsing bad / invalid data.

Reply:
I'm a bit conflicted on this one. Originally I was using tryparse and defaulting to TimeSpan.Zero if it failed to parse, but then I thought it might be better to let the exception bubble up so it is clear that the expiry configuration is incorrect and that documents will not have an expiry set.

I think throwing the exception is pretty standard behaviour, for instance when a string is provided where an integer is expected. I could create a custom exception that makes it more clear what the issue is if you prefer.

Let me know what you want me to do and I'll implement it. For now I have left it.

Reply:
Yeah, I understand your concern here and agree blindly using a default value is also not great behaviour. Lets go with a custom exception to make sure the error is clear to the end user.